diff --git i/server/repository/repository.go w/server/repository/repository.go
index 30ff871..86ff009 100644
--- i/server/repository/repository.go
+++ w/server/repository/repository.go
@@ -292,17 +292,21 @@ func (s *Server) ListApps(ctx context.Context, q *repositorypkg.RepoAppsQuery) (
 	if err := s.enf.EnforceErr(claims, rbac.ResourceRepositories, rbac.ActionGet, createRBACObject(repo.Project, repo.Repo)); err != nil {
 		return nil, err
 	}
+	appProject := q.GetAppProject()
+	if appProject == "" {
+		appProject = repo.Project
+	}
 
 	// This endpoint causes us to clone git repos & invoke config management tooling for the purposes
 	// of app discovery. Only allow this to happen if user has privileges to create or update the
 	// application which it wants to retrieve these details for.
-	appRBACresource := fmt.Sprintf("%s/%s", q.AppProject, q.AppName)
+	appRBACresource := fmt.Sprintf("%s/%s", appProject, q.AppName)
 	if !s.enf.Enforce(claims, rbac.ResourceApplications, rbac.ActionCreate, appRBACresource) &&
 		!s.enf.Enforce(claims, rbac.ResourceApplications, rbac.ActionUpdate, appRBACresource) {
 		return nil, common.PermissionDeniedAPIError
 	}
 	// Also ensure the repo is actually allowed in the project in question
-	if err := s.isRepoPermittedInProject(ctx, q.Repo, q.AppProject); err != nil {
+	if err := s.isRepoPermittedInProject(ctx, q.Repo, appProject); err != nil {
 		return nil, err
 	}
 
@@ -342,9 +346,13 @@ func (s *Server) GetAppDetails(ctx context.Context, q *repositorypkg.RepoAppDeta
 	if err := s.enf.EnforceErr(claims, rbac.ResourceRepositories, rbac.ActionGet, createRBACObject(repo.Project, repo.Repo)); err != nil {
 		return nil, err
 	}
+	appProject := q.GetAppProject()
+	if appProject == "" {
+		appProject = repo.Project
+	}
 	appName, appNs := argo.ParseFromQualifiedName(q.AppName, s.settings.GetNamespace())
 	app, err := s.appLister.Applications(appNs).Get(appName)
-	appRBACObj := createRBACObject(q.AppProject, q.AppName)
+	appRBACObj := createRBACObject(appProject, q.AppName)
 	// ensure caller has read privileges to app
 	if err := s.enf.EnforceErr(claims, rbac.ResourceApplications, rbac.ActionGet, appRBACObj); err != nil {
 		return nil, err
@@ -357,7 +365,7 @@ func (s *Server) GetAppDetails(ctx context.Context, q *repositorypkg.RepoAppDeta
 		}
 	} else {
 		// if we get here we are returning repo details of an existing app
-		if q.AppProject != app.Spec.Project {
+		if appProject != app.Spec.Project {
 			return nil, common.PermissionDeniedAPIError
 		}
 		// verify caller is not making a request with arbitrary source values which were not in our history
@@ -366,7 +374,7 @@ func (s *Server) GetAppDetails(ctx context.Context, q *repositorypkg.RepoAppDeta
 		}
 	}
 	// Ensure the repo is actually allowed in the project in question
-	if err := s.isRepoPermittedInProject(ctx, q.Source.RepoURL, q.AppProject); err != nil {
+	if err := s.isRepoPermittedInProject(ctx, q.Source.RepoURL, appProject); err != nil {
 		return nil, err
 	}
 
@@ -391,7 +399,7 @@ func (s *Server) GetAppDetails(ctx context.Context, q *repositorypkg.RepoAppDeta
 	refSources := make(v1alpha1.RefTargetRevisionMapping)
 	if app != nil && app.Spec.HasMultipleSources() {
 		// Store the map of all sources having ref field into a map for applications with sources field
-		refSources, err = argo.GetRefSources(ctx, app.Spec.Sources, q.AppProject, s.db.GetRepository, []string{})
+		refSources, err = argo.GetRefSources(ctx, app.Spec.Sources, appProject, s.db.GetRepository, []string{})
 		if err != nil {
 			return nil, fmt.Errorf("failed to get ref sources: %w", err)
 		}
